### PR TITLE
Refactor `pkg/generate` package

### PR DESCRIPTION
**Pull request by A-Hilaly**
Issue N/A

During `code-generator` early days, `generate.Generator` used to have
methods that generated controllers, apis and resources code. After few
refactoring iterations the code that outputs controllers Go code is
now situated in `pkg/generate/ack`, and `generate.Generator` only
contains methods to read/load CRDs and Type Definitions.

This PR contains two commits. The first one renames `pkg/generate.Generator`
to `pkg/model.Model` and the second moves `generator` unit tests to `pkg/model`
and `pkg/generate/testdata` to `pkg/testdata`

Description of changes:

- Rename `pkg/generate.Generator` to `pkg/model.Model`, also renames
all the variables and parameters that were used to represent a generator.    
- Move `pkg/generate/testdata` to `pkg/testdata`
- Move all `pkg/generate` unit tests to `pkg/model`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by A-Hilaly** [comment]
NOTE to reviews: please give this PR a manual merge. It will simplify the rebase of #121

**Review by vijtrip2 — APPROVED** [review]

File: pkg/model/crd_util.go [hunk 1]
```diff
@@ -0,0 +1,398 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
```
Comment (RedbackThomson) [inline pkg/model/crd_util.go:1]: What is this entirely new file?

File: pkg/model/type_def_util.go [hunk 2]
```diff
@@ -0,0 +1,226 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
```
Comment (RedbackThomson) [inline pkg/model/type_def_util.go:1]: And this?

File: pkg/model/enum_def.go [hunk 3]
```diff
@@ -56,3 +59,31 @@ func newEnumVal(orig string) EnumValue {
 		Clean:    string(clean),
 	}
 }
+
+// GetEnumDefs takes an *SDKAPI and *ackgenconfig.Config and returns a slice of
+// pointers to `EnumDef` structs which represent string fields whose value is
+// constrained to one or more specific string values.
+func GetEnumDefs(SDKAPI *SDKAPI, cfg *ackgenconfig.Config) ([]*EnumDef, error) {
```
Comment (RedbackThomson) [inline pkg/model/enum_def.go:66]: And this entirely new method?

**Review by RedbackThomson — COMMENTED** [review]
Wondering why this refactor PR has a bunch of new files and methods in it. Also do you know if the generator is referenced in other documentation (such as in `community`) or in the Godocs, that might need changing? I understand doing a search all for "generator" could return a lot of false positives, but I worry that a change this large may introduce a fragmented understanding of the code.

**Comment by A-Hilaly** [comment]
@RedbackThomson This will not impact the existing godocs, since we are keeping the same methods for `Inferrer/Generator` - new methods will be added the `pkg/model` godoc and that's it.

File: pkg/model/crd_util.go (hunk 1 above)
Comment (A-Hilaly) [reply pkg/model/crd_util.go:1]: Just for organization and make navigation easier. `crd.go` for `type CRD` and it's methods; `crd_util.go` for utility functions that interacts with `CRD`s. Also godoc is not generated per go file but per go package - so having multiple files or a single one doesn't impact the generated docs.

File: pkg/model/type_def_util.go (hunk 2 above)
Comment (A-Hilaly) [reply pkg/model/type_def_util.go:1]: for the same reasons

File: pkg/model/enum_def.go (hunk 3 above)
Comment (A-Hilaly) [reply pkg/model/enum_def.go:66]: To keep the the `generate.go` as small as possible and allow other components to call `GetEnumDefs` without having to instantiate a `Generator/Inferrer`

**Comment by A-Hilaly** [comment]
/hold

reworking the refactor

**Comment by A-Hilaly** [comment]
@RedbackThomson i ushed new refactoring commits conserving the `CRD`/`TypeDef` methods

**Comment by RedbackThomson** [comment]
New refactoring is that you've moved `Generator` inside the `model` package and it's now called `Model`? Is there more that I am missing?

**Comment by A-Hilaly** [comment]
>New refactoring is that you've moved Generator inside the model package and it's now called Model? Is there more that I am missing?

Yes exactly, also moved generator unit tests under `pkg/model/model_*_test.go`, and `pkg/generate/testdata` to `pkg/testdata`

**Review by jaypipes — APPROVED** [review]
Yes, excellent.

**Review by RedbackThomson — APPROVED** [review]
Agree this makes a lot more sense. Great!

**Pull request merged**